# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/wgpu-hal/src/gles/device.rs
# Commit: b3109ad14685
# Full Hash: b3109ad14685624ada7861083ff5332aec9a7818
# Author: Andy Leiserson <aleiserson@mozilla.com>
# Date: 2025-10-07 17:00 +0000
# Description:
#   Bug 1992842 - Update wgpu to upstream 1072b878 (2025-10-06) r=webgpu-reviewers,supply-chain-reviewers,ErichDonGubler
# ==============================================================================

diff -r 1d3f000cf1f9 -r b3109ad14685 third_party/rust/wgpu-hal/src/gles/device.rs
--- a/third_party/rust/wgpu-hal/src/gles/device.rs	Tue Oct 07 16:51:44 2025 +0000
+++ b/third_party/rust/wgpu-hal/src/gles/device.rs	Tue Oct 07 17:00:40 2025 +0000
@@ -1564,7 +1564,7 @@
         &self,
         fence: &super::Fence,
         wait_value: crate::FenceValue,
-        timeout_ms: u32,
+        timeout: Option<core::time::Duration>,
     ) -> Result<bool, crate::DeviceError> {
         if fence.satisfied(wait_value) {
             return Ok(true);
@@ -1578,7 +1578,9 @@
         let timeout_ns = if cfg!(any(webgl, Emscripten)) {
             0
         } else {
-            (timeout_ms as u64 * 1_000_000).min(!0u32 as u64)
+            timeout
+                .map(|t| t.as_nanos().min(u32::MAX as u128) as u32)
+                .unwrap_or(u32::MAX)
         };
         fence.wait(gl, wait_value, timeout_ns)
     }